lib/ansible/modules/cloud/vmware/vmware_guest_network.py

```diff
@@ -462,8 +462,19 @@ def get_network_config_spec(self, vm_obj, network_list):
                                 nic_device.connectable.connected = network['connected']
                                 self.change_detected = True
                             if 'name' in network and nic_device.deviceInfo.summary != network['name']:
-                                nic_device.deviceInfo.summary = network['name']
-                                self.change_detected = True
+                                pg_obj = self.find_network_by_name(network_name=network['name'])[0]
```

Comment:
```suggestion
                                pg_obj = self.find_network_by_name(network_name=network['name'])
                                if not pg_obj:
                                	self.module.fail_json(msg="Failed to find network with name %s" % network['name'])
                                pg_obj = pg_obj[0]
                                if isinstance(pg_obj, vim.Network):
                                    nic_device.backing = vim.vm.device.VirtualEthernetCard.NetworkBackingInfo()
                                    nic_device.backing.deviceName = network['name']
                                    nic_device.backing.network = pg_obj
                                 elif isinstance(pg_obj, vim.dvs.DistributedVirtualPortgroup):
                                    nic_device.backing = vim.vm.device.VirtualEthernetCard.DistributedVirtualPortBackingInfo()
                                    nic_device.device.backing.port = vim.dvs.PortConnection()
                                    nic_device.backing.port.switchUuid = pg_obj.config.distributedVirtualSwitch.uuid
                                    nic_device.backing.port.portgroupKey = pg_obj.key
                                 self.change_detected = True
                           
```

Reply:
Thank you for the review and suggested.
The `PortGroup` existence check is found on line [395](https://github.com/ansible/ansible/blob/414d392f3f9f1b4deed010f5d78278aadc29b754/lib/ansible/modules/cloud/vmware/vmware_guest_network.py#L395).
So, it seems to overlap with the following error handling.

```
                                if not pg_obj:
                                    self.module.fail_json(msg="Failed to find network with name %s" % network['name'])
```

What do you think?